hugo/content/docs/smodel.md

```diff
@@ -0,0 +1,554 @@
+
+Abstract class for edges.
+
+*Properties:*
+
+* `routerKind: string` - *optional*: The kind of router to use for the edge.
+* `routingPoints: Point[]` - *optional*: The routing points of the edge. Defaults to an empty array.
+* `sourceId: string`: The id of the source element.
+* `targetId: string`: The id of the target element.
+* `sourceAnchorCorrection: number` - *optional*: The correction of the source anchor. Defaults to `0`.
+* `targetAnchorCorrection: number` - *optional*: The correction of the target anchor. Defaults to `0`.
+
+*Inheritance:*
+
+[`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+## Classes Inheritance
+
+{{< mermaid class="text-center">}}
+flowchart TD;
+SModelElementImpl
+SParentElementImpl
+SChildElementImpl
+SModelRootImpl
+ModelIndexImpl
+SGraphImpl
+ViewportRootElement
+SNodeImpl
+SConnectableElementImpl
+SShapeElementImpl
+SPortImpl
+SEdgeImpl
+SRoutableElementImpl
+SLabelImpl
+SCompartmentImpl
+SGraphIndex
+
+SParentElementImpl --> SModelElementImpl
+SChildElementImpl --> SParentElementImpl
+SModelRootImpl --> SParentElementImpl
+SGraphImpl --> ViewportRootElement
+ViewportRootElement --> SModelRootImpl
+SNodeImpl --> SConnectableElementImpl
+SConnectableElementImpl --> SShapeElementImpl
+SShapeElementImpl --> SChildElementImpl
+SPortImpl --> SConnectableElementImpl
+SEdgeImpl --> SRoutableElementImpl
+SRoutableElementImpl --> SChildElementImpl
+SLabelImpl --> SShapeElementImpl
+SCompartmentImpl --> SShapeElementImpl
+SGraphIndex --> ModelIndexImpl
+{{< /mermaid>}}
+
+## Features
+
+Features can be viewed as a set of functionalities that a given element can support. They can enrich the behavior and usability of the diagram model. SModelElements can have default features that will apply to all model elements that are instance of this SModelElement.
+
+Features can be added in a inherited class by setting the `DEFAULT_FEATURES` field. This way, the default features will be enabled for all instances of the class.
+
+Let's have a look at the `SNodeImpl` class and how it implements default features:
+
+```typescript
+export class SNodeImpl extends SConnectableElementImpl implements Selectable, Fadeable, Hoverable {
+    static readonly DEFAULT_FEATURES = [connectableFeature, deletableFeature, selectFeature, boundsFeature,
+        moveFeature, layoutContainerFeature, fadeFeature, hoverFeedbackFeature, popupFeature];
+    ...
+}
+```
+
+It is possible to fine-tune the behavior in the [dependency injection container](../dependency_injection/#the-container) by enabling or disabling features for a given model element type.
+
+```typescript
+configureElement('my-node-type', SNodeImpl, RectangularNodeView, {enable: [layoutableChildFeature], disable: [moveFeature]})
+```
+
+To ensure that the required additional properties are present on the model element, Sprotty comes with a set of interfaces known as `SModelExtension`.
```

Comment:
I did not get the idea: Sprotty does a check, if all properties required by features are set on the model elements? All feature implementations need to inherit from SModelExtension? ...